chain/ethereum/src/data_source.rs

```diff
@@ -1173,6 +1174,30 @@ impl MappingEventHandler {
         self.topic0
             .unwrap_or_else(|| string_to_h256(&self.event.replace("indexed ", "")))
     }
+
+    pub fn matches(&self, log: &Log) -> bool {
+        if let Some(topic0) = log.topics.get(0) {
+            return self.topic0() == *topic0
```

Comment:
I think it would be great if we could turn these repeated conditions into something with a name.